ui/src/status/actions/index.ts

```diff
@@ -0,0 +1,77 @@
+// he is a library for safely encoding and decoding HTML Entities
+import he from 'he'
+
+import {fetchJSONFeed as fetchJSONFeedAJAX} from 'src/status/apis'
+
+import {notify} from 'src/shared/actions/notifications'
+import {notifyJSONFeedFailed} from 'src/shared/copy/notifications'
+
+import {JSONFeedData} from 'src/types'
+import {AxiosResponse} from 'axios'
+
+export enum ActionTypes {
+  FETCH_JSON_FEED_REQUESTED = 'FETCH_JSON_FEED_REQUESTED',
+  FETCH_JSON_FEED_COMPLETED = 'FETCH_JSON_FEED_COMPLETED',
+  FETCH_JSON_FEED_FAILED = 'FETCH_JSON_FEED_FAILED',
+}
+
+interface FetchJSONFeedRequestedAction {
+  type: ActionTypes.FETCH_JSON_FEED_REQUESTED
+}
+
+interface FetchJSONFeedCompletedAction {
+  type: ActionTypes.FETCH_JSON_FEED_COMPLETED
+  payload: {data: JSONFeedData}
+}
+
+interface FetchJSONFeedFailedAction {
+  type: ActionTypes.FETCH_JSON_FEED_FAILED
+}
+
+export type Action =
+  | FetchJSONFeedRequestedAction
+  | FetchJSONFeedCompletedAction
+  | FetchJSONFeedFailedAction
+
+const fetchJSONFeedRequested = (): FetchJSONFeedRequestedAction => ({
+  type: ActionTypes.FETCH_JSON_FEED_REQUESTED,
+})
+
+const fetchJSONFeedCompleted = (
+  data: JSONFeedData
+): FetchJSONFeedCompletedAction => ({
+  type: ActionTypes.FETCH_JSON_FEED_COMPLETED,
+  payload: {data},
+})
+
+const fetchJSONFeedFailed = (): FetchJSONFeedFailedAction => ({
+  type: ActionTypes.FETCH_JSON_FEED_FAILED,
+})
+
+export const fetchJSONFeedAsync = (url: string) => async (
```

Comment:
wanna try typing the return here? i've been doing it in dashboard actions, and it's a little complicated -- happy to help think this through if you decide to type it